### PR TITLE
SCMOD-13948:  Using opensuse:latest (15.3) requires dirmngr package in builder image to install gosu

### PR DESCRIPTION
**Pull request by DavidGerald-Milligan**
https://portal.digitalsafe.net/browse/SCMOD-13948

### CODE REVIEW

File: release-notes-2.4.0.md [hunk 1]
```diff
@@ -4,5 +4,6 @@
 ${version-number}
 
 #### New Features
+- SCMOD-13948: dirmngr package is required when using latest opensuse-leap 15.3.
```
Comment (michael-bryson) [inline release-notes-2.4.0.md]: What exactly is the dirmngr package required for? It is something to do with the keyserver so add a little more detail on why we need it. Also state that it was removed from the OS packages as leap 15.2 had it pre-installed.

**Review by michael-bryson — CHANGES_REQUESTED** [review]

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
If this is only required for installing `gosu` then make this a multi-stage build and download gosu in the first stage.

File: src/main/docker/Dockerfile [hunk 2]
```diff
@@ -15,25 +15,36 @@
...
+ARG DOCKER_HUB_PUBLIC
+FROM ${DOCKER_HUB_PUBLIC}/opensuse/leap:latest
+
+# Set the locale manually to a set inherited from the base image (defaults to POSIX)
+ENV LANG=en_US.utf8
+
+# Update the OS packages, install postgreSQL client and dejavu-fonts
+RUN zypper -n refresh && \
+    zypper -n update && \
+    zypper -n install postgresql dejavu-fonts && \
```
Comment (michael-bryson) [inline src/main/docker/Dockerfile]: Curl was here before the Gosu install so can you put that back in here so the built image has it?

File: src/main/docker/Dockerfile [hunk 3]
```diff
@@ -15,25 +15,36 @@
...
 RUN gpg --batch --keyserver-options http-proxy=${env.HTTP_PROXY} --keyserver hkps://keys.openpgp.org \
       --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4 && \
     curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/1.12/gosu-amd64" && \
     curl -o /usr/local/bin/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/1.12/gosu-amd64.asc" && \
-    gpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu && \
-    rm /usr/local/bin/gosu.asc && \
-    chmod +x /usr/local/bin/gosu
+    gpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu
+
+ARG DOCKER_HUB_PUBLIC
```
Comment (michael-bryson) [inline src/main/docker/Dockerfile:33]: I don't think this needs repeated.

File: src/main/docker/Dockerfile [hunk 4]
```diff
@@ -15,25 +15,36 @@
...
+
+# Update the OS packages, install postgreSQL client and dejavu-fonts
+RUN zypper -n refresh && \
+    zypper -n update && \
+    zypper -n install postgresql dejavu-fonts && \
+    zypper -n clean --all
+
+# Copy gosu
+COPY --from=builder /usr/local/bin/gosu /usr/local/bin/gosu
+RUN chmod +x /usr/local/bin/gosu
```
Comment (dermot-hardy) [inline src/main/docker/Dockerfile]: Do this in the builder image before copying it over.

File: src/main/docker/Dockerfile (hunk 3 above)
Comment (michael-bryson) [inline src/main/docker/Dockerfile]: Is this needed here?

**Comment by michael-bryson** [comment]
Can you add some comments to the dockerfile indicating where the builder image is being used and the actual image build starts? For example: https://github.com/CAFapi/opensuse-java11-images/blob/develop/src/main/docker/Dockerfile.jre

**Review by michael-bryson — APPROVED** [review]

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**